### PR TITLE
feat: support raw Vue SSR

### PR DESCRIPTION
**Pull request by JohannesLamberts**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

I did not set up a Nuxt app yet, but tested with a basic Vue SSR application. I'm happy to double-check if the Nuxt Plugin still works in case the PR looks good otherwise :)

### CODE REVIEW

**Comment by JohannesLamberts** [comment]
@posva I can get back to the MR next week to resolve the codecov issues. Would you accept setting thresholds in jest.config.js to get faster feedback on coverage?

**Comment by posva** [comment]
Thank you for this! I haven’t had the time yet to check it but I will try to this week or next week

File: __tests__/ssr/app/entry-server.ts [hunk 1]
```diff
@@ -1,8 +1,16 @@
+import Vue from 'vue'
 import { createApp } from './main'
+import { PiniaSsr, getRootState } from '../../../src'
+
+// simulate server-side environment
+// @ts-ignore
+delete global.window
```
Comment (posva) [inline __tests__/ssr/app/entry-server.ts]: we should be able to add 

```js
/**
 * @jest-environment node
 */
```

to the top of the `spec.ts` file to deactivate jsdom

File: src/ssrPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { VueConstructor } from 'vue/types'
+import { setActiveReq } from './rootStore'
+
+export const PiniaSsr = (vue: VueConstructor) => {
+  const isServer = typeof window === 'undefined'
+
+  if (!isServer) {
+    console.warn(
+      'PiniaSsrPlugin seems to be included in the browser bundle. Try to Vue.use(PiniaSsr) only from entry-server.js'
```
Comment (posva) [inline src/ssrPlugin.ts]:
```suggestion
      '`PiniaSsrPlugin` seems to be used in the browser bundle. You should only call it on the server entry: https://github.com/posva/pinia#raw-vue-ssr'
```

File: __tests__/ssr/app/entry-server.ts (hunk 1 above)
Comment (JohannesLamberts) [reply __tests__/ssr/app/entry-server.ts]: thanks :) didn't know that

File: src/ssrPlugin.ts (hunk 2 above)
Comment (JohannesLamberts) [reply src/ssrPlugin.ts]: applied

**Pull request merged**